internal/repository/ent/subscription.go

```diff
@@ -330,6 +330,40 @@ func (r *subscriptionRepository) List(ctx context.Context, filter *types.Subscri
 	return result, nil
 }
 
+// ListActiveSubscriptionsDueForRenewal retrieves all active subscriptions that are due for renewal in 24 hours
+func (r *subscriptionRepository) ListSubscriptionsDueForRenewal(ctx context.Context) ([]*domainSub.Subscription, error) {
```

Comment:
For future - can have limits and offset in this to avoid fetching 1000s of records